docs/openapi-docs/src/main/scala/sttp/tapir/docs/openapi/schema/ObjectSchemasForEndpoints.scala

```diff
@@ -82,14 +81,24 @@ object ObjectSchemasForEndpoints {
 
   private def objectSchemas(typeData: TypeData[_]): List[ObjectTypeData] = {
     typeData match {
-      case TypeData(s @ TSchema(st: TSchemaType.SProduct, _, _, _), validator) =>
-        List(st.info -> TypeData(s, validator): ObjectTypeData) ++ fieldsSchemaWithValidator(st, validator)
-          .flatMap(objectSchemas)
-          .toList
       case TypeData(TSchema(TSchemaType.SArray(o), _, _, _), validator) =>
         objectSchemas(TypeData(o, elementValidator(validator)))
-      case TypeData(s @ TSchema(st: TSchemaType.SCoproduct, _, _, _), validator) =>
-        (st.info -> TypeData(s, validator): ObjectTypeData) +: st.schemas.flatMap(c => objectSchemas(TypeData(c, Validator.pass)))
+      case TypeData(s @ TSchema(st: TSchemaType.SProduct, _, _, _), validator: Validator.Product[_]) =>
+        (st.info -> TypeData(s, validator): ObjectTypeData) +: fieldsSchemaWithValidator(st, validator)
+          .flatMap(objectSchemas)
+          .toList
+      case TypeData(s @ TSchema(st: TSchemaType.SProduct, _, _, _), Validator.CollectionElements(ev: Validator.Product[_], _)) =>
```

Comment:
the case with `Validator.CollectionElements` is a fix for a separate issue, no?

Reply:
Ah, yes, I forgot to update PR's description. Just give me a sec.

Reply:
ah, got it now, makes sense. Wouldn't it be better though to keep the `match` in `fieldsSchemaWithValidator`, and handle the `Validator.Product`/`Validator.CollectionElements` case there? That would decrease the number of top-level cases here, and extract the common logic for products/coproducts

Reply:
There is very little logic to extract so I don't think that it will be that beneficial.

Reply:
I extracted the duplicated logic but I don't think that moving `match` back to `fieldsSchemaWithValidator` is a good idea.

Reply:
👍